Core/src/RequestWrapper.php

```diff
@@ -282,38 +283,44 @@ private function applyHeaders(RequestInterface $request)
         ];
 
         if ($this->shouldSignRequest) {
-            $headers['Authorization'] = 'Bearer ' . $this->getToken();
-        }
+            $quotaProject = $this->quotaProject;
+            $token = null;
 
-        return Psr7\modify_request($request, ['set_headers' => $headers]);
-    }
+            if ($this->accessToken) {
+                $token = $this->accessToken;
+            } else {
+                $credentialsFetcher = $this->getCredentialsFetcher();
+                $token = $this->fetchCredentials($credentialsFetcher);
 
-    /**
-     * Gets the access token.
-     *
-     * @return string
-     */
-    private function getToken()
-    {
-        if ($this->accessToken) {
-            return $this->accessToken;
+                if ($credentialsFetcher instanceof GetQuotaProjectInterface) {
+                    $quotaProject = $credentialsFetcher->getQuotaProject() ?: $quotaProject;
```

Comment:
Does this have the behavior that when `quotaProject` is set in `credentials.json`, it will override the user-supplied `quotaProject` option specified in the Client constructor? If so, I think it should be the other way.

```php
$this->quotaProject = $quotaProject;
if (!$quotaProject && $credentialsFetcher instanceof GetQuotaProjectInterface) {
    $quotaProject = $credentialsFetcher->getQuotaProject();
}
```

Reply:
We actually set the quotaProject into the credentials instance as we create one to use here, so whatever the user passed in would be what we are fetching here.